packages/labs/gen-wrapper-vue/src/lib/wrapper-module-template.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  LitElementDeclaration,
+  ReactiveProperty as ModelProperty,
+  Event as ModelEvent,
+  PackageJson,
+} from '@lit-labs/analyzer/lib/model.js';
+import {
+  javascript,
+  kabobToOnEvent,
+  toInitialCap,
+} from '@lit-labs/gen-utils/lib/str-utils.js';
+
+export const wrapperModuleTemplate = (
+  packageJson: PackageJson,
+  moduleJsPath: string,
+  elements: LitElementDeclaration[]
+) => {
+  return javascript`
+  import { h, defineComponent, openBlock, createBlock } from "vue";
+  import { wrapSlots, Slots, eventProp } from "@lit-labs/vue-utils/wrapper-utils.js";
+  import '${packageJson.name}/${moduleJsPath}';
+
+  ${elements.map((element) => wrapperTemplate(element))}
+`;
+};
+
+// TODO(sorvell): need to extract Javascript type from typescript for Vue.
+// can workaround this by using an SFC and the macro `defineProperties<>()`
+const tsTypeToVuePropType = (type: string) => {
+  const required = type.indexOf('undefined') === -1;
+  let jsType = type.replace('undefined', '').trim().replace(/[|]$/, '');
+  jsType = jsType
+    .split(/\s+/)
+    .map((s) => toInitialCap(s).replace(/[|]/g, '||'))
+    .join(' ')
+    .trim();
+  return `{type: ${jsType}, required: ${required}}`;
+};
+
+const wrapProps = (props: Map<string, ModelProperty>) =>
+  Array.from(props.values())
+    .map(
+      (prop) =>
+        `${prop.name}: ${tsTypeToVuePropType(
+          prop.typeString || 'string|undefined'
+        )}`
+    )
+    .join(',\n');
+
+// TODO(sorvell): Improve event handling, currently just forwarding the event,
+// but this should be its "payload."
```

Comment:
We need a payload concept in Angular too. I think a jsdoc comment on the event type, or some extension to the `@fires` jsdoc might work.

Reply:
Will address via https://github.com/lit/lit/issues/2974.